src/clj_nakadi_java/core.clj

```diff
@@ -47,18 +47,36 @@
     (typeLiteral [_] TypeLiterals/OF_STRING)))
 
 
-(defn ->subscription-stream-config [subscription-id-or-stream-config]
-  (if (string? subscription-id-or-stream-config)
-    (-> (StreamConfiguration.)
-        (.subscriptionId subscription-id-or-stream-config))
-    (do
-      (assert (.subscriptionId subscription-id-or-stream-config)
-              "Stream configuration must include subscriptionId.")
-      subscription-id-or-stream-config)))
-
-
-(defn consume-subscription [client subscription-id-or-stream-config callback]
-  (let [stream-config    (->subscription-stream-config subscription-id-or-stream-config)
+(defn set-param
+  "Sets properties of a StreamConfiguration instance.
+  Takes the value of key from config-map and uses the supplied
+  method to update the instance of StreamConfiguration.
+  Parameters are optional by default."
+  ([StreamConfiguration, config-map, key, method, required]
+   (if-let [value (key config-map)]
+     (method StreamConfiguration value)
+     (do
+       (if required
+         (assert (key config-map) (format "Stream configuration must include %s" key))
+         StreamConfiguration))))
+
+  ([StreamConfiguration, config-map, key, method]
+   (set-param StreamConfiguration config-map key method false)))
+
+
+(defn ->subscription-stream-config [config-map]
+  (-> (StreamConfiguration.)
+      (set-param config-map :subscription-id #(.subscriptionId % %2) true)  ;; required
```

Comment:
I would rather programmatically convert `:subscription-id` to `subscriptionId` method calls to not hardcode specific method here.
Something like this:

```clj
(let [stream-config (StreamConfiguration.)]
  (doseq [[k v] config-map]
    (let [method (convert-keyword-to-method k)] ; should throw an assertion error when method does not exist
      (call-method-via-reflection stream-config method v))
  stream-config)
```

Reply:
Done. However, I decided not to throw an exception if the method does not exist. I feel it is more  resillient this way.